### PR TITLE
validate key and label before creating contact field

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba/flows/tests.py [hunk 1]
```diff
@@ -2047,6 +2047,14 @@ def test_save_to_contact_action(self):
         contact = Contact.objects.get(id=self.contact.pk)
         self.assertEquals("Jen Newcomer", contact.name)
 
+        # throw exception for other reserved words except name and first_name
+        for word in Contact.RESERVED_FIELDS:
```
Comment (norkans7) [inline temba/flows/tests.py]: I added these tests to make sure contact fields we throw anywhere we create contact field, I am not sure how we prevent to use reserved words on the frontend

File: temba/contacts/tests.py [hunk 2]
```diff
@@ -2217,6 +2221,10 @@ def test_manage_fields(self):
         self.assertFormError(response, 'form', None,
                              "Field names can only contain letters, numbers and hypens")
 
+        post_data['label_2'] = 'Name'
+        response = self.client.post(manage_fields_url, post_data, follow=True)
+        self.assertFormError(response, 'form', None, "Field name 'Name' is a reserved word")
+
```
Comment (nicpottier) [inline temba/contacts/tests.py]: Does this error properly appear in the UI?

File: temba/contacts/views.py [hunk 3]
```diff
@@ -1037,7 +1037,7 @@ def clean(self):
                     if label.lower() in used_labels:
                         raise ValidationError(_("Field names must be unique"))
 
-                    elif label in Contact.RESERVED_FIELDS:
+                    elif label.lower() in Contact.RESERVED_FIELDS:
```
Comment (nicpottier) [inline temba/contacts/views.py]: Seems like having a method on ContactField that does a strip / lower and check against RESERVED_FIELDS would be better.. then replacing anywhere we do this kind of validation with that method.

File: temba/contacts/tests.py (hunk 2 above)
Comment (norkans7) [inline temba/contacts/tests.py]: Yes

File: temba/contacts/views.py (hunk 3 above)
Comment (rowanseymour) [inline temba/contacts/views.py]: Shouldn't this be calling `ContactField.is_valid_key(ContactField.make_key(label))` ?

Comment (norkans7) [inline temba/contacts/views.py]: Yep I am changing that

File: temba/contacts/views.py [hunk 4]
```diff
@@ -1034,10 +1034,12 @@ def clean(self):
                     if not ContactField.is_valid_label(label):
                         raise forms.ValidationError(_("Field names can only contain letters, numbers and hypens"))
 
+                    valid_label_gen_key = ContactField.is_valid_key(ContactField.make_key(label))
+
                     if label.lower() in used_labels:
                         raise ValidationError(_("Field names must be unique"))
 
-                    elif label in Contact.RESERVED_FIELDS:
+                    elif not valid_label_gen_key or valid_label_gen_key in Contact.RESERVED_FIELDS:
```
Comment (nicpottier) [inline temba/contacts/views.py]: Still don't understand this.. shouldn't we have a method instead of checking against RESERVED_FIELDS?

Comment (norkans7) [inline temba/contacts/views.py]: Right, that second condition in not needed

**Pull request merged**